packages/commons/src/fhir/task.ts

```diff
@@ -500,3 +500,28 @@ export function notCorrectedHistory(
   }
   return true
 }
+
+export function filterConsecutiveAssignment(
+  task: TaskHistory,
+  index: number,
+  allTasks: TaskHistory[]
+) {
+  const currentAction = getActionFromTask(task)
+
+  if (index === 0) {
+    return true
+  }
+
+  const previousTask = allTasks[index - 1]
+  const previousAction = getActionFromTask(previousTask)
+
+  // Only keep the task if it is not a consecutive ASSIGNED action
+  if (
+    currentAction === TaskAction.ASSIGNED &&
+    previousAction === TaskAction.ASSIGNED
+  ) {
+    return false
+  }
+
+  return true
+}
```

Comment:
Do we need this if the assignment/unassignment works as expected and it doesn't do duplicates? Let's not add it if that is the case, rather migrate the erroneous old data.

Reply:
We can actually have multiple assignment actions consecutively. I think, we discussed about filtering them to keep the latest assignment only.

Reply:
Like if a user logs in from different device, the user will find the previously assigned ones as not-assigned. And then the user assigns the declaration again, consecutive assign actions should appear in the audit page.

Reply:
Hmm yeah, understand. Wonder if we could simply download the record (without assignment) in those cases where you are already assigned to the record? So that we wouldn't need to send assignment audit action and the handling of this could get a bit simpler?

I think this might have been discussed but if you don't mind, let's write it down

Reply:
@jpye-finch If a record is assigned to me, and I re-download it with a separate device, should the re-download be visible in the record's audit log as well? Or just the initial assignment?

Reply:
@naftis @eduffus I think we should be logging if it be assigned to another device. As PII will now be on two different devices. 

We need to consider the copy/messaging to better describe to someone who is viewing the logs that this has happened